### PR TITLE
SOLR-14142 Log in Solr's log whether request logging is enabled

### PR DESCRIPTION
**Pull request by janhoy**
https://issues.apache.org/jira/browse/SOLR-14142

### CODE REVIEW

File: solr/core/src/java/org/apache/solr/util/StartupLoggingUtils.java [hunk 1]
```diff
@@ -187,4 +187,19 @@ public static String getLogLevelString() {
     else if (rootLogger.isErrorEnabled()) return "ERROR";
     else return "INFO";
   }
+
+  /**
+   * Check whether Jetty requestlogging is enabled and log info about it
+   */
+  public static void checkRequestLogging() {
+    boolean requestLogEnabled = System.getProperty("sun.java.command", "").contains("module=requestlog");
```
Comment (janhoy) [inline solr/core/src/java/org/apache/solr/util/StartupLoggingUtils.java:195]: Found no smarter way to check whether that module is active. There is probably some static Jetty class method that would do the same and be more rubust (such as when Oracle changes the name of `sun.java.command` property..?

File: solr/core/src/java/org/apache/solr/core/CoreContainer.java [hunk 2]
```diff
@@ -730,6 +731,8 @@ public void load() {
 
     logging = LogWatcher.newRegisteredLogWatcher(cfg.getLogWatcherConfig(), loader);
 
+    StartupLoggingUtils.checkRequestLogging();
```
Comment (janhoy) [inline solr/core/src/java/org/apache/solr/core/CoreContainer.java:734]: I could have logged this already in `CoreContainerProvider` but I wanted to have the warn included in the Admin UI, and thus had to do it after initializing the `LogWatcher`. I wonder if we can initialize LogWatcher earlier on?

File: solr/core/src/java/org/apache/solr/util/StartupLoggingUtils.java (hunk 1 above)
Comment (madrob) [reply solr/core/src/java/org/apache/solr/util/StartupLoggingUtils.java:195]: If we are executing somewhere that has access to the `Server` object (like the tests `JettySolrRunner`) then we can check if `server.getRequestLog()` returns null. I haven't figured out how to get access to our server object outside of that though. This is what I get when looking at who has a reference to the Server object and none of those seem particularly approachable.

image

Comment (madrob) [reply solr/core/src/java/org/apache/solr/util/StartupLoggingUtils.java:195]: I suspect that being able to arbitrarily load the server object would be some kind of security constraint violation, but I haven't really dug into it much further.

Comment (janhoy) [reply solr/core/src/java/org/apache/solr/util/StartupLoggingUtils.java:195]: Could of course edit bin/solr and bin/solr.cmd to add a custom system property for this (should only be a few extra lines), but I don't think it's worth it. Worst case is that this logging disappears if the string pattern in command line args changes.

File: solr/core/src/java/org/apache/solr/core/CoreContainer.java (hunk 2 above)
Comment (janhoy) [reply solr/core/src/java/org/apache/solr/core/CoreContainer.java:734]: Spun off #543 for earlier init

File: solr/core/src/java/org/apache/solr/util/StartupLoggingUtils.java (hunk 1 above)
Comment (janhoy) [reply solr/core/src/java/org/apache/solr/util/StartupLoggingUtils.java]: Ok, I found a trick, where I could set a sysprop in `solr/server/etc/jetty-requestlog.xml` by some xml tags, so now I check on that instead.

**Comment by janhoy** [comment]
I think this should be ready to merge now.

File: solr/core/src/java/org/apache/solr/util/StartupLoggingUtils.java [hunk 3]
```diff
@@ -187,4 +187,20 @@ public static String getLogLevelString() {
     else if (rootLogger.isErrorEnabled()) return "ERROR";
     else return "INFO";
   }
+
+  /**
+   * Check whether Jetty request logging is enabled and log info about it.
+   * The property "solr.log.requestlog.enabled is set in solr/server/etc/jetty-requestlog.xml
+   */
+  public static void checkRequestLogging() {
+    boolean requestLogEnabled = System.getProperty("solr.log.requestlog.enabled", "false").equals("true");
```
Comment (madrob) [inline solr/core/src/java/org/apache/solr/util/StartupLoggingUtils.java:196]:
```suggestion
    boolean requestLogEnabled = Boolean.getBoolean("solr.log.requestlog.enabled");
```

**Comment by janhoy** [comment]
Thanks for the review. I'll merge now.

**Pull request merged**